pkg/backup/backup_job.go

```diff
@@ -1899,15 +1903,36 @@ func (b *backupResumer) maybeNotifyScheduledJobCompletion(
 			// Not a scheduled backup.
 			return nil
 		}
-
-		scheduleID := jobspb.ScheduleID(tree.MustBeDInt(datums[0]))
+		scheduleID = jobspb.ScheduleID(tree.MustBeDInt(datums[0]))
 		if err := jobs.NotifyJobTermination(ctx, txn, env, b.job.ID(), jobState, b.job.Details(), scheduleID); err != nil {
 			return errors.Wrapf(err,
 				"failed to notify schedule %d of completion of job %d", scheduleID, b.job.ID())
 		}
 		return nil
-	})
-	return err
+	}); err != nil {
+		return err
+	}
+
+	if scheduleID != 0 {
+		var backupStmt string
+		if err := execCtx.ExecCfg().InternalDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
```

Comment:
nit: i think we should put this compaction job creation code in it's own helper func. That way, `maybeNotifyScheduledJobCompletion` can warn on any error returned by the helper. We don't want any logic related to compaction to send an error to the resumer.

Perhaps we also rename `maybeNotifyScheduledJobCompletion` to `processScheduledBackupCompletion`